### PR TITLE
Type declarations for redux-persist-webextension-storage

### PR DESCRIPTION
**Pull request by martonlederer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: types/redux-persist-webextension-storage/redux-persist-webextension-storage-tests.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import _reduxPersistWebextensionStorage = require("redux-persist-webextension-storage");
```
Comment (peterblazejewicz) [inline types/redux-persist-webextension-storage/redux-persist-webextension-storage-tests.ts:1]:
```suggestion
import { localStorage, syncStorage } from 'redux-persist-webextension-storage';
```

File: types/redux-persist-webextension-storage/redux-persist-webextension-storage-tests.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+_reduxPersistWebextensionStorage.localStorage();
+_reduxPersistWebextensionStorage.syncStorage();
```
Comment (peterblazejewicz) [inline types/redux-persist-webextension-storage/redux-persist-webextension-storage-tests.ts:4]:
```suggestion
localStorage();
syncStorage();
```

File: types/redux-persist-webextension-storage/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface StorageInterface {
+  getItem(key: string): Promise<any>;
+  removeItem(key: string): Promise<void>;
+  setItem(key: string, value: any): Promise<void>;
+}
+
+export function localStorage(): StorageInterface;
+export function syncStorage(): StorageInterface;
```
Comment (peterblazejewicz) [inline types/redux-persist-webextension-storage/index.d.ts:13]: are those funcs or instances (constants)?
https://github.com/ssorallen/redux-persist-webextension-storage/blob/b6cd619b3a627335e996402e681108c26b7edb04/src/createStorage.js#L3

File: types/redux-persist-webextension-storage/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+export {};
```
Comment (peterblazejewicz) [inline types/redux-persist-webextension-storage/index.d.ts:14]:
```suggestion
```
You don't need this one, just use `export` keyword for `localStorage`, `syncStorage` and `StorageInterface` (unless the intent is to not export `StorageInterface`)

Comment (martonlederer) [reply types/redux-persist-webextension-storage/index.d.ts:14]: No I don't want to export that, it's just a helper

Comment (peterblazejewicz) [reply types/redux-persist-webextension-storage/index.d.ts:14]: cool, so just disregard this comment, thx!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martonlederer thx!

**Comment by orta** [comment]
👍🏻 same

**Pull request merged**